app/src/main/java/com/navapp/navigation/util/math/TSP/HelpKarpSolver.java

```diff
@@ -0,0 +1,112 @@
+package com.navapp.navigation.util.math.TSP;
+
+import com.navapp.navigation.util.math.MathLocation;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+//This class is based on the code from the Heuristics-TSP library by Sinclert Pérez, @Sinclert on Github, see Heuristics-TSP-license.md
+public class HelpKarpSolver<T extends MathLocation> implements TSPSolver<T> {
```

Comment:
Why implement yourself instead of using something existing?

Didn't exactly read it, but this might be nice: https://developers.google.com/optimization/routing/tsp#java